staff/views/staff.py

```diff
@@ -169,4 +169,53 @@ def slide_delete(request, pk):
         messages.success(request, 'slide has been deleted')
         return redirect('staff:slide_list')
 
+
 # end static info section
+
+
+def create_settings(request):
+    user = UserProfile.objects.get(user__username=request.user.username)
+    allowed = ['superadmin', 'admin']
+    if user.user_type not in allowed:
+        return render(request, 'staff/permissiondenied.html')
+    else:
+        if len(OllySetting.objects.all()) > 1:
+            # one already exists
+            messages.warning(request, 'Settings already exist, redirecting to modify current settings.')
+            return redirect('staff:edit_settings')
+        else:
+            if request.method == 'POST':
+                # validate data, make object
+                form = CreateOllySetting(request.POST)
+                if form.is_valid():
+                    temp = form.instance
+                    temp.save()
+                    #temp = OllySetting()
+                    #temp.freeze_team_invites = form.cleaned_data['freeze_team_invites']
+                    #temp.disable_team_creation = form.cleaned_data['disable_team_creation']
+                    #temp.save()
+                    messages.success(request, "Settings created")
+                    return redirect('staff:edit_settings')
+            else:
+                form = CreateOllySetting()
+                return render(request, 'staff/pages/create_olly_settings.html', {'form': form})
+
+
+def edit_settings(request):
+    user = UserProfile.objects.get(user__username=request.user.username)
+    allowed = ['superadmin', 'admin']
+    if user.user_type not in allowed:
+        return render(request, 'staff/permissiondenied.html')
+    else:
```

Comment:
Another unnecessary branch here - this `else` only serves to nest the code and make it harder to follow.